shared/AppInsightsCore/Tests/Unit/src/CookieManager.Tests.ts

```diff
@@ -749,6 +774,513 @@ export class CookieManagerTests extends AITestClass {
+                });
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: Dynamic config change with maxAge parameter should preserve options",
+            test: () => {
+                // Create a dynamic config that can be updated
+                let configValues = {
+                    cookieCfg: objExtend({}, this._cookieMgrCfg, {
+                        enabled: false
+                    })
+                };
+                let manager = createCookieMgr(configValues);
+
+                let newKey = "test." + newId();
+                let newValue = newId();
+                let maxAge = 3600; // 1 hour
+
+                // Set a cookie with maxAge while disabled - should be cached
+                let result = manager.set(newKey, newValue, maxAge);
+                Assert.equal(true, result, "Set should return true even when disabled (cached)");
+                Assert.equal(undefined, this._testCookies[newKey], "Cookie should not be in storage when disabled");
+                Assert.equal(newValue, manager.get(newKey), "Should return cached value");
+
+                // Enable cookies via dynamic config change - should flush cached value with maxAge
+                configValues.cookieCfg.enabled = true;
+
+                // Cookie should be flushed with the maxAge parameter
+                let expectedValue = this.isEmulatingIe ? 
+                    newValue + "; path=/; expires=" : 
+                    newValue + "; path=/; expires=";
+                Assert.ok(this._testCookies[newKey].indexOf(expectedValue) === 0, "Cookie should be flushed with expires");
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: Dynamic config change from legacy disableCookiesUsage should flush cached values",
+            test: () => {
+                // Create a dynamic config that can be updated
+                let configValues = {
+                    disableCookiesUsage: true,
+                    cookieCfg: this._cookieMgrCfg
+                };
+                let manager = createCookieMgr(configValues);
+
+                let newKey = "test." + newId();
+                let newValue = newId();
+
+                // Set a cookie while disabled - should be cached
+                let result = manager.set(newKey, newValue);
+                Assert.equal(true, result, "Set should return true even when disabled (cached)");
+                Assert.equal(undefined, this._testCookies[newKey], "Cookie should not be in storage when disabled");
+
+                // Get should return the cached value
+                Assert.equal(newValue, manager.get(newKey), "Should return cached value when disabled");
+
+                // Enable cookies via dynamic config change - should flush cached values
+                configValues.disableCookiesUsage = false;
+
+                // Verify the cookie was flushed to actual storage
```

Comment:
You need to add a clock.tick(1) after changing the state to allow the dynamic configurations changes to take affect